<a href="https://colab.research.google.com/github/abhilasha8/play_store_app_review_analysis/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

## <b> Import the Libraries

In [29]:
# Import the required libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt,seaborn as sns
%matplotlib inline

In [30]:
# mount the drive
from google.colab import drive
drive.mount('/content/Drive')

Drive already mounted at /content/Drive; to attempt to forcibly remount, call drive.mount("/content/Drive", force_remount=True).


## <b> Read the data into data frame

In [31]:
path = '/content/Drive/My Drive/Colab Notebooks/PYTHON/Capstone Project - Play Store app Review/'
ps_df = pd.read_csv(path + 'Play Store Data.csv')
ur_df = pd.read_csv(path + 'User Reviews.csv')

## <b> Analyze the Play store app data

In [33]:
# Display the play store app data
ps_df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [34]:
# Check the shape of data
ps_df.shape

(10841, 13)

In [35]:
# All columns name in the data frame
ps_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [36]:
# Information about the data frame
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [37]:
# Statistical Description of data frame
ps_df.describe(include='all')

App Category  ...         Current Ver Android Ver
count    10841    10841  ...               10833       10838
unique    9660       34  ...                2832          33
top     ROBLOX   FAMILY  ...  Varies with device  4.1 and up
freq         9     1972  ...                1459        2451
mean       NaN      NaN  ...                 NaN         NaN
std        NaN      NaN  ...                 NaN         NaN
min        NaN      NaN  ...                 NaN         NaN
25%        NaN      NaN  ...                 NaN         NaN
50%        NaN      NaN  ...                 NaN         NaN
75%        NaN      NaN  ...                 NaN         NaN
max        NaN      NaN  ...                 NaN         NaN

[11 rows x 13 columns]

There are 10841 rows and 13 columns in play store app data frame, dataset seems to have missing values most of which occurs for the Rating column, all the other column except Rating are in object datatype. So, the clean up is required.

## <b> Data Cleaning

1. Handling the NULL values

In [38]:
# Find the NULL values
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [39]:
# Find the number of missing value in each column
ps_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

There is large number of NULL values in the Rating column, so we will drop all the NULL values.

In [40]:
# Drop the missing value in rating column
ps_df.dropna(subset=['Rating'],axis=0, inplace = True)

In [42]:
# Check the shape of the data frame after dropping the null values
ps_df.shape

(9367, 13)

In [43]:
# recalculate the number of null values
ps_df.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    1
Genres            0
Last Updated      0
Current Ver       4
Android Ver       3
dtype: int64

Since, there are few numbers of NULL values, so replace the NULL values with the mode values.

In [44]:
# replace the null values
ps_df['Android Ver'].fillna(ps_df['Android Ver'].mode()[0],inplace=True)
ps_df['Current Ver'].fillna(ps_df['Current Ver'].mode()[0],inplace=True)
ps_df['Content Rating'].fillna(ps_df['Content Rating'].mode()[0],inplace=True)

In [45]:
# Final check for the missing values
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9367 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9367 non-null   object 
 1   Category        9367 non-null   object 
 2   Rating          9367 non-null   float64
 3   Reviews         9367 non-null   object 
 4   Size            9367 non-null   object 
 5   Installs        9367 non-null   object 
 6   Type            9367 non-null   object 
 7   Price           9367 non-null   object 
 8   Content Rating  9367 non-null   object 
 9   Genres          9367 non-null   object 
 10  Last Updated    9367 non-null   object 
 11  Current Ver     9367 non-null   object 
 12  Android Ver     9367 non-null   object 
dtypes: float64(1), object(12)
memory usage: 1.0+ MB


2. Handling the outliers

In [47]:
# Check the outliers
ps_df[ps_df.Rating > 5]

App Category  ...  Current Ver Android Ver
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9  ...   4.0 and up  4.1 and up

[1 rows x 13 columns]

By looking at above row we can say that there is something wrong with the above row, because installs can't be free, type can't be 0, price can't be everyone and moreover, genres can't be a date. So, we have to decide the drop that row.

In [48]:
# Drop the row with wrong values
ps_df.drop([10472],inplace=True)

3. Handling the dulicate files

In [53]:
# Check if there is any duplicate row or not
ps_df[ps_df.duplicated()]

App  ...         Android Ver
229                       Quick PDF Scanner + OCR FREE  ...        4.0.3 and up
236                                                Box  ...  Varies with device
239                                 Google My Business  ...          4.4 and up
256                                ZOOM Cloud Meetings  ...          4.0 and up
261                          join.me - Simple Meetings  ...          4.4 and up
...                                                ...  ...                 ...
8643                    Wunderlist: To-Do List & Tasks  ...  Varies with device
8654   TickTick: To Do List with Reminder, Day Planner  ...  Varies with device
8658                           ColorNote Notepad Notes  ...  Varies with device
10049        Airway Ex - Intubate. Anesthetize. Train.  ...          5.0 and up
10768                                             AAFP  ...          5.0 and up

[474 rows x 13 columns]

There are 474 rows with duplicate entries, so we have to dropped them.

In [70]:
# Drop the duplicate files
ps_df.drop_duplicates(inplace = True)

# Check the shape of the data frame after dropping the dulicate files
ps_df.shape

(8892, 13)